pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -430,6 +430,7 @@ func (r *ReconcileHyperConverged) ensureHcoDeleted(req *common.HcoRequest) (reco
 
 	// Remove the finalizer
 	req.Instance.ObjectMeta.Finalizers = drop(req.Instance.ObjectMeta.Finalizers, FinalizerName)
+	req.Instance.ObjectMeta.Finalizers = drop(req.Instance.ObjectMeta.Finalizers, badFinalizerName)
 	req.Dirty = true
```

Comment:
I think we maybe need to requeue only if the finilizer were actually removed. Somethinkg like:
```golang
        if contains(req.Instance.ObjectMeta.Finalizers, FinalizerName) {
	    req.Instance.ObjectMeta.Finalizers = drop(req.Instance.ObjectMeta.Finalizers, FinalizerName)
            req.Dirty = true
        }
	if contains(req.Instance.ObjectMeta.Finalizers, badFinalizerName) {
	    req.Instance.ObjectMeta.Finalizers = drop(req.Instance.ObjectMeta.Finalizers, badFinalizerName)
            req.Dirty = true
        }
```

Or maybe change the `drop` function to also return boolean `dropped`, and use this result instead of checking again as suggested above.

Reply:
done